basepath_test.go

```diff
@@ -140,3 +141,53 @@ func TestNestedBasePaths(t *testing.T) {
 		}
 	}
 }
+
+
+func TestBasePathOpenFile(t *testing.T) {
+	baseFs := &MemMapFs{}
+	baseFs.MkdirAll("/base/path/tmp", 0777)
+	bp := NewBasePathFs(baseFs, "/base/path")
+	f, err := bp.OpenFile("/tmp/file.txt", os.O_CREATE, 0600)
+	if err != nil {
+		t.Fatalf("failed to open file: %v", err)
+	}
+	if filepath.Dir(f.Name()) != filepath.Clean("/tmp") {
+		t.Fatalf("realpath leaked: %s", f.Name())
+	}
+}
+
+func TestBasePathCreate(t *testing.T) {
+	baseFs := &MemMapFs{}
+	baseFs.MkdirAll("/base/path/tmp", 0777)
+	bp := NewBasePathFs(baseFs, "/base/path")
+	f, err := bp.Create("/tmp/file.txt")
+	if err != nil {
+		t.Fatalf("failed to create file: %v", err)
+	}
+	if filepath.Dir(f.Name()) != filepath.Clean("/tmp") {
+		t.Fatalf("realpath leaked: %s", f.Name())
+	}
+}
+
+func TestBasePathTempFile(t *testing.T) {
+	baseFs := &MemMapFs{}
+	baseFs.MkdirAll("/base/path/tmp", 0777)
+	bp := NewBasePathFs(baseFs, "/base/path")
+
+	tDir, err := TempDir(bp, "/tmp", "")
+	if err != nil {
+		t.Fatalf("Failed to TempDir: %v", err)
+	}
+	fmt.Println(tDir)
```

Comment:
debugging output left in the test

Reply:
my bad, just saw this! removed the println.